src/h5web/utils.ts

```diff
@@ -9,21 +9,27 @@ import type { NxAttribute } from './vis-packs/nexus/models';
 
 export const formatValue = format('.3~e');
 export const formatPreciseValue = format('.5~e');
-export function formatComplex(value: H5WebComplex, specifier = '') {
-  const [real, imag] = value;
-  const formatFunction = format(specifier);
+export const formatMatrixValue = format('.3e');
+export const formatMatrixComplex = createComplexFormatter('.2e', true);
+export const formatScalarComplex = createComplexFormatter('.12~g');
 
-  if (imag === 0) {
-    return `${formatFunction(real)}`;
-  }
+export function createComplexFormatter(specifier: string, forceFull = false) {
+  const formatVal = format(specifier);
```

Comment:
Another benefit of the refactoring: the formatter is now created only once instead of for every value.